src/hbase-2.5/hbase_region_inspector/hbase/impl.clj

```diff
@@ -0,0 +1,108 @@
+(ns hbase-region-inspector.hbase.impl
+  (:require [clojure.set :as set]
+            [hbase-region-inspector.util :as util]
+            [hbase-region-inspector.hbase.base :as base])
+  (:import [org.apache.hadoop.hbase
+            util.Bytes client.Admin client.ConnectionFactory
+            HRegionInfo RegionLoad ClusterStatus ServerName ServerLoad]
+           java.nio.ByteBuffer))
+
+;; http://archive.cloudera.com/cdh5/cdh/5/hbase-0.98.6-cdh5.3.3/apidocs/index.html
+
+(defn info->map
+  "Builds map from HRegionInfo"
+  [^HRegionInfo info]
+  (assoc (base/info->map info)
+         :table (str (.getTable info))))
+
+(defn load->map
+  "Builds map from RegionLoad"
+  [^RegionLoad load]
+  (let [base {:compacted-kvs              (.getCurrentCompactedKVs load)
+              :memstore-size-mb           (.getMemStoreSizeMB load)
+              :read-requests              (.getReadRequestsCount load)
+              :requests                   (.getRequestsCount load)
+              :root-index-size-kb         (.getRootIndexSizeKB load)
+              :store-file-index-size-mb   (.getStorefileIndexSizeMB load)
+              :store-files                (.getStorefiles load)
+              :store-file-size-mb         (.getStorefileSizeMB load)
+              :stores                     (.getStores load)
+              ; :store-uncompressed-size-mb
+              :total-compacting-kvs       (.getTotalCompactingKVs load)
+              :bloom-size-kb              (.getTotalStaticBloomSizeKB load)
+              :total-index-size-kb        (.getTotalStaticIndexSizeKB load)
+              :write-requests             (.getWriteRequestsCount load)}
+        loc  (.getDataLocality load)]
+    (assoc base
+      :store-uncompressed-size-mb (.getStoreUncompressedSizeMB load)
+      :locality (* 100 loc)
+      :local-size-mb (* loc (:store-file-size-mb base)))))
+
+(defn server-load->map
+  "Transforms ServerLoad object into clojure map"
+  [^ServerLoad load]
+  (assoc
+    {:max-heap-mb        (.getMaxHeapMB load)
+     :used-heap-mb       (.getUsedHeapMB load)
+     :regions            (.getNumberOfRegions load)
+     :requests-rate      (.getNumberOfRequests load)
+     :store-files        (.getStorefiles load)
+     :store-file-size-mb (.getStorefileSizeInMB load)}
+    :store-uncompressed-size-mb
+    (some->> load
+             str
+             (re-find #"storefileUncompressedSizeMB=([0-9]+)")
+             last
+             Integer/parseInt)))
```

Comment:
Looks like we can use `ServerLoad#getStoreUncompressedSizeMB()` instead of doing this.

IIRC, I used this method back then because HBase 0.94 didn't provide the API.